src/main/java/ay2021s1_cs2103_w16_3/finesse/logic/commands/EditCommand.java

```diff
@@ -109,10 +109,8 @@ private static Transaction createEditedTransaction(Transaction transactionToEdit
 
         if (transactionToEdit instanceof Expense) {
             return new Expense(updatedTitle, updatedAmount, updatedDate, updatedCategories);
-        } else if (transactionToEdit instanceof Income) {
-            return new Income(updatedTitle, updatedAmount, updatedDate, updatedCategories);
         } else {
-            return new Transaction(updatedTitle, updatedAmount, updatedDate, updatedCategories);
+            return new Income(updatedTitle, updatedAmount, updatedDate, updatedCategories);
```

Comment:
```suggestion
            assert (transactionToEdit instanceof Income);
            return new Income(updatedTitle, updatedAmount, updatedDate, updatedCategories);
```

Since the previous assertion was removed, maybe an assertion here might be necessary? Same for similar occurrences.